### PR TITLE
#162948832 Configure Postgres as Database Engine

### PR DESCRIPTION
**Pull request by andrewhingah**
**What does this PR do?**

Configures Postgres as the db engine for Authors Heaven

**Description of tasks to be completed?**

Configure Postgres as the db engine for Authors Heaven.

**How should this be manually tested?**

- Install Postgres on your computer
- Clone this repo `git clone https://github.com/andela/ah-django-unchained.git`
- Create a virtual environment `virtualenv venv`
- Create a `.env` file and add the following db configurations:
```
source venv/bin/activate
export DB_NAME = "your_db_name"
export DB_USER = "your_postgres_username"
export DB_PASS = "your_postgres_password"
export DB_HOST = "localhost or any other host name"
export DB_PORT = "port_number"
```
- Source your environment variables `source .env`
- Install dependencies `pip install -r requirements.txt`

**Setting Up the Database**
- Start Postgres server `pg_ctl -D /usr/local/var/postgres start`
- To open database on terminal use `psql Postgres`
- Create a database with name `auths_heaven` 
- Migrate changes `python manage.py makemigrations` then `python manage.py migrate`

If the table models are created in the database then the database connections is successful

**Relevant Pivotal Tracker Stories**

#162948832

### CODE REVIEW

**Comment by kwameasiago** [comment]
Looks good.Remember to squash your commits

**Comment by KMaina** [comment]
Good work, however I see that you have branched off `master` branch. Should it not be `develop`?

**Comment by andrewhingah** [comment]
> Good work, however I see that you have branched off `master` branch. Should it not be `develop`?

Hey Ken, the instructions indicate that we should raise PR to master

**Comment by MaggieChege** [comment]
Nice work. Instead of using psycopg2 use psycopg2-binary.

**Comment by andrewhingah** [comment]
> Looks good.Remember to squash your commits



> Looks good.Remember to squash your commits

Hey Kwame. Why do you think squashing commits here is necessary?

**Comment by Georgeygigz** [comment]
> Nice work. Instead of using psycopg2 use psycopg2-binary.
Okay, thanks we are going to work on that

File: authors/settings.py [hunk 1]
```diff
@@ -81,8 +81,13 @@
...
 DATABASES = {
     'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'auths_heaven',
+        'USER': 'postgres',
+        'PASSWORD': 'DB_PASS',
+        'HOST': 'localhost',
+        'PORT': '5432',
```
Comment (ignatiusukwuoma) [inline authors/settings.py]: We do not want to expose these details in our repo. Create a .env file and hide them there (all the keys except ENGINE). Let user know they need to add these keys (with values) to their own .env file, so you won't need to tell them what NAME or USER or PORT to use. 
All user needs to do is set up a Postgres database and add their details to their .env file. `.env` should be gitignored

Comment (andrewhingah) [reply authors/settings.py]: Well noted.

**Comment by d-kahara** [comment]
> Nice work. Instead of using psycopg2 use psycopg2-binary.

Since we'll be using a version of Django ≥1.9, the DB engine should be defined as django.db.backends.postgresql

**Comment by ignatiusukwuoma** [comment]
Let's make it easy for a developer to export the db configurations by only putting the key=values into a .env file instead of running `export` everytime the server is to be run. You can use this package https://github.com/theskumar/python-dotenv to load the content of  the `.env` file into the environment. 

I also want to see the `Database Setup` sub-section under the `Installation` section in the Readme containing the database set-up information

**Review by ignatiusukwuoma — APPROVED** [review]

**Pull request merged**